media-libs/libjpeg-turbo/libjpeg_turbo-1.5.3.recipe

```diff
@@ -0,0 +1,139 @@
+SUMMARY="MMX, SSE, and SSE2 SIMD accelerated JPEG library"
+DESCRIPTION="libjpeg-turbo is a JPEG image codec that uses SIMD instructions \
+(MMX, SSE2, AVX2, NEON, AltiVec) to accelerate baseline JPEG compression and \
+decompression on x86, x86-64, ARM, and PowerPC systems. On such systems, \
+libjpeg-turbo is generally 2-6x as fast as libjpeg, all else being equal. On \
+other types of systems, libjpeg-turbo can still outperform libjpeg by a \
+significant amount, by virtue of its highly-optimized Huffman coding routines.
+In many cases, the performance of libjpeg-turbo rivals that of proprietary \
+high-speed JPEG codecs.
+libjpeg-turbo implements both the traditional libjpeg API as well as the less \
+powerful but more straightforward TurboJPEG API. libjpeg-turbo also features \
+colorspace extensions that allow it to compress from/decompress to 32-bit and \
+big-endian pixel buffers (RGBX, XBGR, etc.), as well as a full-featured Java \
+interface.
+libjpeg-turbo was originally based on libjpeg/SIMD, an MMX-accelerated \
+derivative of libjpeg v6b developed by Miyasaka Masaru. The TigerVNC and \
+VirtualGL projects made numerous enhancements to the codec in 2009, and in \
+early 2010, libjpeg-turbo spun off into an independent project, with the goal \
+of making high-speed JPEG compression/decompression technology available to \
+a broader range of users and developers."
+HOMEPAGE="https://www.libjpeg-turbo.org/"
+COPYRIGHT="1991-2018 Thomas G. Lane, Guido Vollbeding
+	1997-2010 Guido Vollbeding
+	2000-2017 D. R. Commander
+	2009 Bill Allombert
+	1999-2006 MIYASAKA Masaru
+	2011 Siarhei Siamashka
+	2013 Linaro Limited
+	2014 MIPS Technologies, Inc.
+	2015 Google, Inc.
+	2015 Matthieu Darbois"
+LICENSE="BSD (3-clause)
+	IJG
+	Zlib"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/libjpeg-turbo/libjpeg-turbo-$portVersion.tar.gz"
+CHECKSUM_SHA256="b24890e2bb46e12e72a79f7e965f409f4e16466d00e1dd15d93d73ee6b592523"
+SOURCE_DIR="libjpeg-turbo-$portVersion"
+PATCHES="libjpeg_turbo-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64 arm"
```

Comment:
```diff
-ARCHITECTURES="x86_gcc2 x86 x86_64 arm"
+ARCHITECTURES="?x86_gcc2 ?x86 ?x86_64 ?arm"
```
I'd recommend marking all architectures as untested because when I restarted my x86_gcc2 system after having replaced `jpeg{,_tools,_devel}` by `libjpeg_turbo{,_tools,_devel}` Tracked was not able to start.
(I did not try rebooting my other Haiku installs (x86 and x86_64) so we might not have the problem on these architectures.

Anyway, unless someone investigates this problem on each arch, I think we should mark everything as untested.

Reply:
It's expected. `jpeg` is a dependency of `haiku`, and since `libjpeg_turbo` provides different soversion, it's not a surprise that things wouldn't work. To fix this, we would need to rebuild Haiku with `libjpeg_turbo`.

Reply:
OK, thanks.
Well, I think it is safer to stick to jpeg until the core developers decide to switch to libjpeg_turbo.

Reply:
Now I've got to think if we even have to declare this package as conflicts with `jpeg`, given the different soname they carry.

For `_devel`, however, `CONFLICTS` still remains necessary.

Reply:
Sorry @alaviss, I had not noticed your comment.
You are right that both packages don't overlap.
FWIW:

package | jpeg | libjpeg_turbo
-- | -- | --
symlink | libjpeg.so.9 | libjpeg.so.62
shared lib | libjpeg.so.9.3.0 | libjpeg.so.62.2.0

Anyway, it might be safer to keep libjpeg_turbo marked as untested until everyone is OK with enabling it.

Reply:
I would say we enable it and drop the conflicts for `jpeg`. This way, we would allow one to build pkgs with `libjpeg_turbo` and install it alongside their `jpeg`-based Haiku.

Reply:
Well, anyone willing to install libjpeg_turbo can do it locally. It's easy to do for those users who know what they are doing.

I'd say that having this small barrier is nice, because if anyone could install libjpeg_turbo without being aware of the possible consequences, then we would end up with 2 groups of users: those who have switched, and those who have not.
In such a world, whenever we need to test a package which depends on libjpeg, we would need to check that it's OK with both the _standard_ libjpeg (from the jpeg package) and with the alternate libjpeg (from libjpeg_turbo).

So I think we should only mark libjpeg_turbo as tested and drop the conflicts for its base package after one user, at least, has had the chance to rebuild (and test) everything against the libjpeg from libjpeg_turbo. Of course, we would also need to make sure Haiku's core developers are OK with the switch.

There's something else we need to check: will libjpeg_turbo build on arm and ppc? If not, maybe we should tweak libjpeg_turbo's pkconfig files and install libjpeg with an alternate name, so that packages which only build on `x86{,_gcc2,_x86}` and depend on libjpeg would be able to **_opt-in_** the dependency on libjpeg_turbo.

Now, I'm not against a switch. I just wish we don't do that before we are sure it won't break anything.
Regading the advances on Haiku for arm & ppc, I hope the possible libjpeg switch will not be an issue for these architectures.